resource_handler.go

```diff
@@ -101,5 +101,6 @@ type ResourceHandler interface {
 	Delete(r *http.Request, id string) error
 	// Patch update one or more attributes of a SCIM resource using a sequence of
 	// operations to "add", "remove", or "replace" values.
+	// If you return empty Resource.Attributes, 204 No Content will be response.
```

Comment:
Added an explanation comment!

Reply:
We should give a more detailed explanation here. On which occasions they can return no attributes (e.g. no changes made). Everything should be clear after reading the comment.

If you want me to write the comment, let me know. I can be picky on these things. 😄

Reply:
Could you write the comments for me?
Thank you!

Reply:
Alright, I'll write one later today!

Reply:
```go
// ...
//
// If you return no Resource.Attributes, a 204 No Content status code will be returned.
// This case is only valid in the following scenarios:
// 1. the Add/Replace operation should return No Content only when the value already exists AND is the same.
// 2. the Remove operation should return No Content when the value to be remove is already absent.
// More information in Section 3.5.2 of RFC 7644: https://tools.ietf.org/html/rfc7644#section-3.5.2
```